### PR TITLE
Issue/620 transcribe undoCastlingFischerRandom test in jest

### PR DESCRIPTION
**Pull request by ChiragAgg5k**
Closes issue #620 .

- added helper functions
- added chromedriver module
- added dev dependency for @babel/preset-env (my ide kept giving me error without it, let me know if its required)

### CODE REVIEW

File: src/__tests__/automated/undoCastlinginFischerRandomGame.spec.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  afterAll(async () => {
+    stopDriver(driver);
+  });
+
+  it('Undo castling in a Fischer Random game', async () => {
+    await driver.get("https://www.chesslablab.com:9443/");
+    await driver.manage().window().setRect({ width: 1366, height: 742 });
+
+    await driver.findElement(By.id("Nav-analysisBoard")).click();
```
Comment (programarivm) [inline src/__tests__/automated/undoCastlinginFischerRandomGame.spec.js:19]:
```
npm test -- undoCastlinginFischerRandomGame.spec.js
```

```
  console.debug
    Selenium Manager binary found at /home/standard/projects/react-chess/node_modules/selenium-webdriver/bin/linux/selenium-manager

      at getBinary (node_modules/selenium-webdriver/common/seleniumManager.js:56:13)

  console.debug
    Driver path: /home/standard/projects/react-chess/node_modules/.bin/chromedriver

      at logOutput (node_modules/selenium-webdriver/common/seleniumManager.js:146:15)

  console.debug
    Browser path: /usr/bin/google-chrome

      at logOutput (node_modules/selenium-webdriver/common/seleniumManager.js:146:15)

 FAIL  src/__tests__/automated/undoCastlinginFischerRandomGame.spec.js
  Undo castling in a Fischer Random game
    ✕ Undo castling in a Fischer Random game (1097 ms)

  ● Undo castling in a Fischer Random game › Undo castling in a Fischer Random game

    NoSuchElementError: no such element: Unable to locate element: {"method":"css selector","selector":"*[id="Nav-analysisBoard"]"}
      (Session info: headless chrome=117.0.5938.132)

      17 |     await driver.manage().window().setRect({ width: 1366, height: 742 });
      18 |
    > 19 |     await driver.findElement(By.id("Nav-analysisBoard")).click();
         |     ^
      20 |     await driver.findElement(By.id("Nav-analysisBoard-MenuItem-fenString")).click();
      21 |     await driver.findElement(By.id("LoadFenDialog-TextField-fen")).click();
      22 |     await driver.findElement(By.id("LoadFenDialog-TextField-fen")).sendKeys("qnrnbkrb/pppppppp/8/8/8/8/PPPPPPPP/QNRNBKRB w KQkq -");

      at Object.throwDecodedError (node_modules/selenium-webdriver/lib/error.js:524:15)
      at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:601:13)
      at Executor.execute (node_modules/selenium-webdriver/lib/http.js:529:28)
      at thenableWebDriverProxy.execute (node_modules/selenium-webdriver/lib/webdriver.js:745:17)
      at toWireValue (node_modules/selenium-webdriver/lib/webdriver.js:145:15)
      at node_modules/selenium-webdriver/lib/webdriver.js:195:16
      at forEachKey (node_modules/selenium-webdriver/lib/webdriver.js:189:9)
      at convertKeys (node_modules/selenium-webdriver/lib/webdriver.js:194:3)
      at thenableWebDriverProxy.execute (node_modules/selenium-webdriver/lib/webdriver.js:743:22)
      at Object.<anonymous> (src/__tests__/automated/undoCastlinginFischerRandomGame.spec.js:19:5)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        5.075 s, estimated 7 s
Ran all test suites matching /undoCastlinginFischerRandomGame.spec.js/i.

Active Filters: filename /undoCastlinginFischerRandomGame.spec.js/
 › Press c to clear filters.

Watch Usage
 › Press a to run all tests.
 › Press f to run only failed tests.
 › Press o to only run tests related to changed files.
 › Press q to quit watch mode.
 › Press i to run failing tests interactively.
 › Press p to filter by a filename regex pattern.
 › Press t to filter by a test name regex pattern.
 › Press Enter to trigger a test run.
```

**Review by programarivm — CHANGES_REQUESTED** [review]

File: src/__tests__/automated/undoCastlinginFischerRandomGame.spec.js (hunk 1 above)
Comment (ChiragAgg5k) [reply src/__tests__/automated/undoCastlinginFischerRandomGame.spec.js:19]: I faced the same error on my local machine...

The id name is same idk why its giving error

Comment (programarivm) [reply src/__tests__/automated/undoCastlinginFischerRandomGame.spec.js]: This should do the trick.

```js
await driver.wait(until.elementLocated(By.id("Nav-analysisBoard"))).click();
``` 

See https://github.com/chesslablab/react-chess/pull/648

Keep it up, and happy learning!

**Comment by ChiragAgg5k** [comment]
added the fix

File: src/__tests__/automated/undoCastlinginFischerRandomGame.spec.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  beforeAll(async () => {
+    driver = getDriver();
+  });
+
+  afterAll(async () => {
+    stopDriver(driver);
+  });
+
+  it('Undo castling in a Fischer Random game', async () => {
```
Comment (programarivm) [inline src/__tests__/automated/undoCastlinginFischerRandomGame.spec.js:15]:
```
npm test -- undoCastlinginFischerRandomGame.spec.js
```

```
  console.debug
    Selenium Manager binary found at /home/standard/projects/react-chess/node_modules/selenium-webdriver/bin/linux/selenium-manager

      at getBinary (node_modules/selenium-webdriver/common/seleniumManager.js:56:13)

  console.debug
    Driver path: /home/standard/projects/react-chess/node_modules/.bin/chromedriver

      at logOutput (node_modules/selenium-webdriver/common/seleniumManager.js:146:15)

  console.debug
    Browser path: /usr/bin/google-chrome

      at logOutput (node_modules/selenium-webdriver/common/seleniumManager.js:146:15)

 FAIL  src/__tests__/automated/undoCastlinginFischerRandomGame.spec.js (6.346 s)
  Undo castling in a Fischer Random game
    ✕ Undo castling in a Fischer Random game (5001 ms)

  ● Undo castling in a Fischer Random game › Undo castling in a Fischer Random game

    thrown: "Exceeded timeout of 5000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      13 |   });
      14 |
    > 15 |   it('Undo castling in a Fischer Random game', async () => {
         |   ^
      16 |     await driver.get("https://www.chesslablab.com:9443/");
      17 |     await driver.manage().window().setRect({ width: 1366, height: 742 });
      18 |     await driver.wait(until.elementLocated(By.id("Nav-analysisBoard"))).click();

      at src/__tests__/automated/undoCastlinginFischerRandomGame.spec.js:15:3
      at Object.<anonymous> (src/__tests__/automated/undoCastlinginFischerRandomGame.spec.js:4:1)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        7.281 s, estimated 8 s
Ran all test suites matching /undoCastlinginFischerRandomGame.spec.js/i.

Active Filters: filename /undoCastlinginFischerRandomGame.spec.js/
 › Press c to clear filters.

Watch Usage
 › Press a to run all tests.
 › Press f to run only failed tests.
 › Press o to only run tests related to changed files.
 › Press q to quit watch mode.
 › Press i to run failing tests interactively.
 › Press p to filter by a filename regex pattern.
 › Press t to filter by a test name regex pattern.
 › Press Enter to trigger a test run.
```

**Review by programarivm — CHANGES_REQUESTED** [review]

**Comment by programarivm** [comment]
Thank you @ChiragAgg5k for elaborating on the testing issues! It seems as though transcribing this test into Jest is not too obvious. Would you want to help @ShatilKhan and try to solve https://github.com/chesslablab/react-chess/issues/616 instead? 

See https://github.com/chesslablab/react-chess/issues/618

Keep it up,

**Comment by ChiragAgg5k** [comment]
Yeah, i will look into that.

**Pull request closed without merging**